src/framework/Internal/Commands/TestMethodCommand.cs

```diff
@@ -74,14 +76,43 @@ public override TestResult Execute(TestExecutionContext context)
         private object RunTestMethod(TestExecutionContext context)
         {
 #if NET_4_5
+            if (MethodHelper.IsFSharpAsyncMethod(testMethod.Method))
+                return RunFSharpAsyncMethod(context);
+
             if (MethodHelper.IsAsyncMethod(testMethod.Method))
                 return RunAsyncTestMethod(context);
-            else
 #endif
-                return RunNonAsyncTestMethod(context);
+            return RunNonAsyncTestMethod(context);
         }
 
 #if NET_4_5
+        object RunFSharpAsyncMethod(TestExecutionContext context)
+        {
+            object res = Reflect.InvokeMethod(testMethod.Method, context.TestObject, arguments);
+
+            try
+            {
+                var parameters = new[]
+                    {
+                        res,
+                        context.TestCaseTimeout > 0
+                            ? Microsoft.FSharp.Core.FSharpOption<int>.Some(context.TestCaseTimeout)
```

Comment:
Based on my comment below, I don't think we should be including language specific code in the framework.